pkg/store/copr/region_cache.go

```diff
@@ -26,11 +27,233 @@ import (
 	derr "github.com/pingcap/tidb/pkg/store/driver/error"
 	"github.com/pingcap/tidb/pkg/store/driver/options"
 	"github.com/pingcap/tidb/pkg/util/logutil"
+	"github.com/pingcap/tidb/pkg/util/redact"
 	"github.com/tikv/client-go/v2/metrics"
 	"github.com/tikv/client-go/v2/tikv"
 	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 )
 
+// Helper functions for logging
+func formatLocation(loc *tikv.KeyLocation) zap.Field {
+	return zap.Object("location", zapcore.ObjectMarshalerFunc(func(enc zapcore.ObjectEncoder) error {
+		enc.AddUint64("regionID", loc.Region.GetID())
+		enc.AddString("startKey", redact.Key(loc.StartKey))
+		enc.AddString("endKey", redact.Key(loc.EndKey))
+		if loc.Buckets != nil {
+			enc.AddInt("bucketCount", len(loc.Buckets.Keys))
+		}
+		return nil
+	}))
+}
+
+func keyField(name string, key []byte) zap.Field {
+	return zap.String(name, redact.Key(key))
+}
+
+func formatRanges(ranges *KeyRanges) zap.Field {
+	return zap.Object("ranges", zapcore.ObjectMarshalerFunc(func(enc zapcore.ObjectEncoder) error {
+		enc.AddInt("count", ranges.Len())
+		if ranges.Len() > 0 {
+			enc.AddString("firstStart", redact.Key(ranges.At(0).StartKey))
```

Comment:
Just the first of the start range and last of the end range, is it enough for debugging?